### PR TITLE
Featurerequest ctest

### LINKED ISSUE #100: Unit tests and continuous integration
We need to be more serious about testing and breaking stuff especially in core and engine as more people start to use it. Unit tests are the way to go.
- [ ] Write unit tests for core features
- [x]  Run a CI script daily on master

### PR DESCRIPTION
**Pull request by nmellado**
* **Please check if the PR fulfills these requirements**
- [x] Is the Pull-Request done against the right branch:
  - `master`: for hotfixes not impacting the API
  - `master-v1`: for changes related to the Radium Stable Release V1.
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Switch to CTest for testing + add tests in CI

* **What is the current behavior?** (You can also link to an open issue here)
Tests are compiled and ran using home an home made system, and not used in CI.
This PR is expected to close #100.

* **What is the new behavior (if this is a feature change)?**
Tests are now handled using CTest + ran on Travis CI

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Test writing rules have changed.
Also, to make the test compilation working I had to fix a bad design in the buildchain: radium libraries were not including 3rdparty headers explicitly (using `target_include_directories`) but rather implicitly (using `include_directories` in `src/CMakelist.txt`. This was preventing to use include directory transitivity with CMake.

* **Other information**:
It is better if we squash this PR.

### CODE REVIEW

**Comment by nmellado** [comment]
PR ready
Please review.

**Comment by MathiasPaulin** [comment]
Seems good for me but is there a documentation on how to program unitary tests when we add a new functionality in Radium libs ?. It could be helpfull, at least for an example of extending the ``Radium::Core``, and referenced in the Radium programming manual.

File: cmake/ConfigureTesting.cmake [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    add_test(NAME test_${_option_TARGET}
+             #CONFIGURATIONS Release
+             COMMAND test_${_option_TARGET})
+    target_link_libraries(test_${_option_TARGET} ${_option_LIBS})
+    target_include_directories(test_${_option_TARGET} PUBLIC ${testing_include_dir})
+endfunction()
+
+
+# check whether /bin/bash exists
+find_file(RADIUM_BIN_BASH_EXISTS "/bin/bash" PATHS "/" NO_DEFAULT_PATH)
```
Comment (hoshiryu) [inline cmake/ConfigureTesting.cmake]: is it used anywhere?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -154,7 +150,9 @@ endif ( RADIUM_IO_IS_INTERFACE )
 
 
 add_dependencies( ${io_target} ${core_target} ${engine_target} )
-target_include_directories(${io_target} PUBLIC ${RADIUM_SRC_DIR})
+target_include_directories(${io_target}
+    PUBLIC ${RADIUM_SRC_DIR}
```
Comment (hoshiryu) [inline src/CMakeLists.txt]: no need for Eigen?

Comment (MathiasPaulin) [reply src/CMakeLists.txt]: We do not necessarily need it. Actually this is add by each IO support (eg ``AssimpLoader/CMakeLists.txt``
I think that each IO-support must define the minimal set of include directory, that is not the case actually.  See commit about that.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -191,10 +189,13 @@ add_library(
     ${engine_shaders}
 )
 
-target_include_directories( ${engine_target}
+target_include_directories(${engine_target}
+    PUBLIC ${RADIUM_SRC_DIR}
```
Comment (MathiasPaulin) [inline src/CMakeLists.txt]:
```suggestion
    PUBLIC ${RADIUM_SRC_DIR}
           ${EIGEN3_INCLUDE_DIR}
```

**Pull request merged**